bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosVariable.java

```diff
@@ -0,0 +1,414 @@
+     */
+    public boolean isOk() {
+        boolean check;
+
+        // this.access has one of the allowed values
+        check = (this.access.equals(HeliosVariable.ACCESS_R)) || (this.access.equals(HeliosVariable.ACCESS_W))
+                || (this.access.equals(HeliosVariable.ACCESS_RW));
+
+        // this.type has one of the allowed values
+        check = check && ((this.type.equals(HeliosVariable.TYPE_STRING))
+                || (this.type.equals(HeliosVariable.TYPE_INTEGER)) || (this.type.equals(HeliosVariable.TYPE_FLOAT)));
+
+        // this.minValue and this.maxValue are either not set or minValue is less than maxValue
+        check = check && (((this.minVal == null) && (this.maxVal == null)) || (this.minVal <= this.maxVal));
+
+        // length is set
+        check = check && (this.length > 0);
+
+        // count is set
+        check = check && (this.count > 0);
+
+        return check;
+    }
+
+    /**
+     * Checks if the variable has write access
+     *
+     * @return true if the variable has write access
+     */
+    public boolean hasWriteAccess() {
+        return (this.access.equals(HeliosVariable.ACCESS_W)) || (this.access.equals(HeliosVariable.ACCESS_RW));
+    }
+
+    /**
+     * Checks if the variable has read access
+     *
+     * @return true if the variable has read access
+     */
+    public boolean hasReadAccess() {
+        return (this.access.equals(HeliosVariable.ACCESS_R)) || (this.access.equals(HeliosVariable.ACCESS_RW));
+    }
+
+    /**
+     * Checks if the provided value is within the accepted range
+     *
+     * @param value The value as a string
+     * @return true if the value is within the accepted range
+     */
+    public boolean isInAllowedRange(String value) {
+        if ((this.getMinVal() != null) && (this.getMaxVal() != null)) { // min and max value are set
+            try {
+                if (this.type.equals(HeliosVariable.TYPE_INTEGER)) {
+                    // using long becuase some variable are specified with a max of 2^32-1
+                    // parsing double to allow floating point values to be processed as well
+                    Long l = new Double(value).longValue();
+                    return (this.getMinVal().longValue() <= l) && (this.getMaxVal().longValue() >= l);
+                } else if (this.type.equals(HeliosVariable.TYPE_FLOAT)) {
+                    Double d = Double.parseDouble(value);
+                    return (this.getMinVal() <= d) && (this.getMaxVal() >= d);
+                }
+            } catch (Exception e) {
+                return false;
+            }
+        } else {
+            return true; // no range to check
+        }
+        return false;
+    }
+
+    @Override
+    public int compareTo(HeliosVariable v) {
+        if (this.getVariable() < v.getVariable()) {
+            return -1;
+        } else if (this.getVariable() == v.getVariable()) {
+            return 0;
+        } else {
+            return 1;
+        }

```

Comment:
What about this?
```suggestion
        return getVariable() - v.getVariable();
```

Reply:
Well, that would be really elegant then... that's not quite my style ;)